packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -25,30 +25,31 @@ import DataViewsSearch from '../dataviews-search';
 import DataViewsViewConfig from '../dataviews-view-config';
 import { normalizeFields } from '../../normalize-fields';
 import type { Action, Field, View, SupportedLayouts } from '../../types';
-import type { SelectionOrUpdater } from '../../private-types';
+import type { SelectionOrUpdater, SetSelection } from '../../private-types';
 
 type ItemWithId = { id: string };
```

Comment:
@youknowriad could we accept item `id` being a Number?

```suggestion
type ItemWithId = { id: string | number };
```

Reply:
I'll defer to @oandregal and others. But do you want the "id" to be a number? If your ids are number in your objects, you can always provide `getItemId` prop and return `String( item.id )`

Reply:
Also how does this relate to the current PR?

Reply:
Just wondering because when the IDs are `string[]`, the getItemId gets optional. The fixture we use for the stories, the IDs are `numbers []`, making the `getItemId()` not optional.